src/url-setters.cpp

```diff
@@ -4,19 +4,21 @@
 
 namespace ada {
 
-  void url::set_username(const std::string_view input) {
-    if (cannot_have_credentials_or_port()) return;
+  bool url::set_username(const std::string_view input) {
+    if (cannot_have_credentials_or_port()) { return false; }
     username = ada::unicode::percent_encode(input, character_sets::USERINFO_PERCENT_ENCODE);
+    return true;
   }
 
-  void url::set_password(const std::string_view input) {
-    if (cannot_have_credentials_or_port()) return;
+  bool url::set_password(const std::string_view input) {
+    if (cannot_have_credentials_or_port()) { return false; }
     password = ada::unicode::percent_encode(input, character_sets::USERINFO_PERCENT_ENCODE);
+    return true;
   }
 
-  void url::set_port(const std::string_view input) {
-    if (cannot_have_credentials_or_port()) { return; }
-    parse_port(input);
+  bool url::set_port(const std::string_view input) {
+    if (cannot_have_credentials_or_port()) { return false; }
+    return parse_port(input,false);
```

Comment:
I suggest we have a `false` default value for this, to avoid this line.

Reply:
Done.